modules/library/src/main/scala/zio/elasticsearch/ElasticAggregation.scala

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2022 LambdaWorks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package zio.elasticsearch
+
+import zio.elasticsearch.ElasticPrimitive._
+import zio.json.ast.Json
+import zio.json.ast.Json.Obj
+
+sealed trait ElasticAggregation { self =>
+
+  def paramsToJson: Json
+
+  final def toJson: Json = Obj("aggs" -> paramsToJson)
+
+}
+
+object ElasticAggregation {
+
+  def multipleAggregations: MultipleAggregations = MultipleAggregations.empty
```

Comment:
I wonder if instead of having to use `multipleAggregation.aggregations(...)` we should have something like `terms(...).with(otherAgg).sub(subAgg.with(otherSubAgg))`